src/olympia/addons/models.py

```diff
@@ -237,57 +238,137 @@ def not_disabled_by_mozilla(self):
+    ):
+        if types not in (amo.GROUP_TYPE_ADDON, amo.GROUP_TYPE_THEME):
+            raise ImproperlyConfigured(
+                'types needs to be either GROUP_TYPE_ADDON or GROUP_TYPE_THEME'
+            )
+        theme_review = types == amo.GROUP_TYPE_THEME
+        qs = self.get_base_queryset_for_queue(
+            admin_reviewer=admin_reviewer,
+            # The select related needed to avoid extra queries in other queues
+            # typically depend on current_version, but we don't care here, as
+            # it's a pre-review queue. We'll make the select_related() calls we
+            # need ourselves.
+            select_related_fields_for_listed=False,
+            # We'll filter on pending_rejection below without limiting
+            # ourselves to the current_version.
+            exclude_listed_pending_rejection=False,
+            theme_review=theme_review,
+        )
+        filters = (
+            Q(
+                status__in=statuses,
+                type__in=types,
+                versions__channel=amo.RELEASE_CHANNEL_LISTED,
+                versions__files__status=amo.STATUS_AWAITING_REVIEW,
+                versions__reviewerflags__pending_rejection=None,
+            )
+            & ~Q(disabled_by_user=True)
+        )
+        if recommendable:
+            filters &= Q(promotedaddon__group_id=RECOMMENDED.id)
+        elif not theme_review:
+            filters &= ~Q(promotedaddon__group_id=RECOMMENDED.id) & (
+                Q(versions__files__is_webextension=False)
+                | Q(reviewerflags__auto_approval_disabled=True)
+                | Q(reviewerflags__auto_approval_disabled_until_next_approval=True)
+                | Q(reviewerflags__auto_approval_delayed_until__gt=datetime.now())
+                | Q(
+                    promotedaddon__group_id__in=(
+                        group.id for group in PRE_REVIEW_GROUPS
+                    )
+                )
+            )
+
+        return (
+            # We passed select_related_fields_for_listed=False but there are
+            # still base select_related() fields in that queryset that are
+            # applied in all cases that we don't want.
+            qs.select_related(None)
+            .filter(filters)
+            .select_related('reviewerflags')
+            .annotate(
+                first_version_nominated=Min('versions__nomination'),
+                latest_version=Func(F('versions__version'), function='LOWER'),
+            )
+        )
+
+    def get_addons_with_unlisted_versions_queue(self, admin_reviewer=False):
+        qs = self.get_base_queryset_for_queue(
+            select_related_fields_for_listed=False,
+            exclude_listed_pending_rejection=False,
+            admin_reviewer=admin_reviewer,
+        )
+        return (
+            qs.filter(versions__channel=amo.RELEASE_CHANNEL_UNLISTED)
+            .exclude(status=amo.STATUS_DISABLED)
+            .select_related(None)
```

Comment:
what does `.select_related(None)` do?

Reply:
It removes the previous `select_related()` that have been added to the queryset so far by `get_base_queryset_for_queue()`. I'll add a comment.